### DISCUSSION TITLE
Is it supposed to constant number value instead of using Enum value as default value in a intEnum column?

### DISCUSSION THREAD

**Discussion by sprhawk** [simolus3/drift#2784] [General]
I'm using following definition:
```
IntColumn get type => intEnum<ConversationType>()
      .withDefault(Constant(ConversationType.Chat.index))();
```

But it failed to compile, due to undefined ConversationType in generated file.

Is it supposed to use constant as following? 
```
IntColumn get type => intEnum<ConversationType>()
      .withDefault(const Constant(0))();
```

**Comment by simolus3** [comment]
> due to undefined ConversationType in generated file.

Are you defining the table in a file that is different from the database file? If so, try adding an import for `ConversationType` into the database file as well. Unfortunately `part` files can't have their own imports so drift can't add these imports for you.

**Reply by sprhawk** [reply]
Ah, thanks, you are right !

**Comment by sprhawk** [comment]
Another question, the generated schema_versions.dart by:
```
dart run drift_dev schema steps drift_schemas/ lib/models/schema_versions.dart
```
also reported missing my defined enums, is there a workaround for it ?
Thanks

**Reply by simolus3** [reply]
There isn't. The problem is that schema versions reflect an older version of your database, and so it would get pretty confusing if you changed or later removed the `ConversationType` enum (since drift has no control over that).

**Reply by sprhawk** [reply]
Thanks, then I can only use Constant(0) as the default value for it, right ?

**Reply by simolus3** [reply]
Hm right :( That's not ideal, but I don't see an easy way out of this.
